### PR TITLE
Add Gamelogic DTOs

### PR DESCRIPTION
**Pull request by Sikredo**

### CODE REVIEW

**Comment by Sikredo** [comment]
Things to discuss:

- [ ] Should we use a Gamefield Wrapper Class as stated in the UML?

**Comment by raphaelseher** [comment]
|File|Coverage [0%]|:green_apple:|
|:-|:-:|:-:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/2ce4a6f1aa3cf214f89442fb777b289aadb76b5a/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FStartFragment.java)|0%|:green_apple:|

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> Things to discuss:
> 
>     * [ ]  Should we use a Gamefield Wrapper Class as stated in the UML?

Yes sounds good.

**Comment by raphaelseher** [comment]
|File|Coverage [0%]|:green_apple:|
|:-|:-:|:-:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/22920bd325762c898c21f79fb86fe976aaf1ba07/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FStartFragment.java)|0%|:green_apple:|

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

File: gamelogic/src/main/java/at/aau/se2/gamelogic/models/GameField.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package at.aau.se2.gamelogic.models;
+
+import java.util.ArrayList;
+
+public class GameField {
+  private GameFieldRows rows;
```
Comment (raphaelseher) [inline gamelogic/src/main/java/at/aau/se2/gamelogic/models/GameField.java:6]: Ich glaub wir könnten paar Properties immutable machen (final), dann haben wir a Barriere, falls jemand hacken anfängt ;D
Row, cardDecks, heroes z.B.

File: gamelogic/src/main/java/at/aau/se2/gamelogic/models/Hero.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package at.aau.se2.gamelogic.models;
+
+import at.aau.se2.gamelogic.models.heroactions.HeroActionParams;
+
+public class Hero {
+  private int id;
+  private String name;
+  private HeroActionParams heroAction;
+  private int heroActionCoolDown;
```
Comment (raphaelseher) [inline gamelogic/src/main/java/at/aau/se2/gamelogic/models/Hero.java:9]: Cooldown in Zügen, könnten wir kommentieren, damits klar wird.

File: gamelogic/src/main/java/at/aau/se2/gamelogic/models/heroactions/HeroDoDamageParams.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package at.aau.se2.gamelogic.models.heroactions;
+
+import java.util.ArrayList;
+
+public class HeroDoDamageParams extends HeroActionParams {
```
Comment (raphaelseher) [inline gamelogic/src/main/java/at/aau/se2/gamelogic/models/heroactions/HeroDoDamageParams.java:5]: Würd ich auf AttackParams umbennennen, ähnlich wie die CardActions. Unterscheidung von Hero/Card haben wir eh durchs Package

**Comment by raphaelseher** [comment]
|File|Coverage [0%]|:green_apple:|
|:-|:-:|:-:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/8bbaea6f8f83d7556a3f25066e9e73b1a4782d4b/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FStartFragment.java)|0%|:green_apple:|

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
|File|Coverage [0%]|:green_apple:|
|:-|:-:|:-:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/8bbaea6f8f83d7556a3f25066e9e73b1a4782d4b/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FStartFragment.java)|0%|:green_apple:|

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
Boah diese Actions ....
Neue Actions sind auf Develop gemerged und develop ist hier rein gemerged und trotzdem werden alte Actions ausgeführt?

**Comment by raphaelseher** [comment]
|File|Coverage [48.4%]|:green_apple:|
|:-|:-:|:-:|
|[RowType.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FRowType.java)|100%|:green_apple:|
|[InitialPlayer.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FInitialPlayer.java)|100%|:green_apple:|
|[CardDecks.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FCardDecks.java)|87.67%|:green_apple:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|84.44%|:green_apple:|
|[GameField.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameField.java)|65.85%|:green_apple:|
|[GameFieldRows.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameFieldRows.java)|62.75%|:green_apple:|
|[Card.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FCard.java)|38.67%|:green_apple:|
|[Row.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FRow.java)|32.43%|:green_apple:|
|[Player.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FPlayer.java)|27.78%|:green_apple:|
|[Hero.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FHero.java)|0%|:green_apple:|
|[CardType.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FCardType.java)|0%|:green_apple:|
|[RowStatus.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FRowStatus.java)|0%|:green_apple:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FStartFragment.java)|0%|:green_apple:|
|[HeroActionParams.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fheroactions%2FHeroActionParams.java)|0%|:green_apple:|
|[AttackParams.java](https://github.com/SE2-Gwent/Gwent/blob/ea11c28bd4f27a55de660862064d032d851afb86/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fheroactions%2FAttackParams.java)|0%|:green_apple:|

|Total Project Coverage|30.01%|:green_apple:|
|:-|:-:|:-:|

**Comment by Sikredo** [comment]
Schaut so aus wie wenn sie gecached waren, jetzt scheints aber zu funktionieren. Diese Actions sind einfach der Hauptgrund wieso  man nicht GitHub sonder GitLab mit deren CI/CD verwendet :DD

**Pull request merged**